src/test/java/pages/CalculatorPage.java

```diff
@@ -0,0 +1,57 @@
+package pages;
+
+import io.appium.java_client.MobileElement;
+import io.appium.java_client.android.AndroidDriver;
+import io.appium.java_client.pagefactory.AppiumFieldDecorator;
+import org.openqa.selenium.support.FindBy;
+import org.openqa.selenium.support.PageFactory;
+
+import java.time.Duration;
+
+public class CalculatorPage {
+    protected AndroidDriver<MobileElement> driver;
+
+    @FindBy(id = "com.google.android.calculator:id/op_add")
+    private MobileElement sumBtn;
+
+    @FindBy(id = "com.google.android.calculator:id/op_sub")
+    private MobileElement subtractionBtn;
+
+    @FindBy(id = "com.google.android.calculator:id/eq")
+    private MobileElement equalBtn;
+
+    @FindBy(id = "com.google.android.calculator:id/result_final")
+    private MobileElement result;
+
+    public CalculatorPage(AndroidDriver<MobileElement> driver) {
+        this.driver = driver;
+        PageFactory.initElements(new AppiumFieldDecorator(driver, Duration.ofSeconds(30)), this);
+    }
+
+    public void performSum(String ... numbers) {
+        for (String number : numbers) {
+            clickOnDigit(number);
+            if (number.equals(numbers[numbers.length - 1])) {
```

Comment:
that is barely readable, please convert to fori and you'll have access to last elem

Reply:
done